Create a Pull Request template (#59)
ChrisScianna/ROS-Underwater-RnD: docs/pull_request_template.md

# Description

> lease include a summary of the change and which issue is fixed. Also, include motivation and context. List all dependencies required for this change (e.g. links to dependent PRs).

Reviewer: @patapufate nit: 
```suggestion
Please include a summary of the change and which issue is fixed. Also, include motivation and context. List all dependencies required for this change (e.g. links to dependent PRs).
```

Author: fixed.

Fixes # (issue)

## Type of change

> Please mark options that are not relevant.

Reviewer: @patapufate relevant or not relevant?

Author: fixed

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

> # How To Test?

Author: Fixed.

> Please describe the procedure and/or tests that need to be performed to verify these changes. Also, include test configuration details if relevant.

Reviewer: @patapufate nit: 
```suggestion
Please describe the procedure and/or tests that need to be performed to verify these changes work as expected. Also, include test configuration details if relevant.
```

- [ ] Your changes have to be tested on hardware

> - [ ] You have added unit tests

Reviewer: @patapufate I think these two don't add much information. Unit tests must always pass, and the author must verify that (note the testbed isn't a unit tests suite). Tests below may or may not be integration tests on hardware, but it can be specified when/if necessary.

Author: Ok. I'll remov it.

## Test A
1) Step 1
2) Step 2

## Test B
1) Step 1
2) Step 2

> # Checklist:

Reviewer: @patapufate nit: 
```suggestion
# Checklist
```

Reviewer: Also, shall we move this above testing procedures? It looks much more like a checklist for the author than anything else.

Author: Yes. Moved and fixed

- [ ] I have performed a self-review of my own code
- [ ] I have commented my code, particularly in hard-to-understand areas
- [ ] The code builds clean without any errors or warnings

> - [ ] Linter tests are passing

Reviewer: @patapufate 
```suggestion
- [ ] Unit tests are passing
- [ ] Linter tests are passing
```

Author: Added

> - [ ] You have performed linter tests

Reviewer: @patapufate nit:
```suggestion
- [ ] Linter tests are passing
```

Author: Fixed

# Description

> Please include a summary of the change and which issue is fixed. Please also include relevant motivation and context. List any dependencies that are required for this change.

Reviewer: @patapufate 
```suggestion
Please include a summary of the change and which issue is fixed. Also, include motivation and context. List all dependencies required for this change (e.g. links to dependent PRs).
```

Author: Fixed

Fixes # (issue)

## Type of change

> Please delete options that are not relevant.

Reviewer: @patapufate shall we delete, or shall we mark the right options?

Author: we should mark.

- [ ] Bug fix (non-breaking change which fixes an issue)
- [ ] New feature (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)

> # How Has This To Be Tested?

Reviewer: @patapufate nit: 
```suggestion
# How To Test
```

Author: Fixed

> Please describe the tests (Instructions) to verify the changes. Please also list any relevant details for your test configuration

Reviewer: @patapufate nit: 
```suggestion
Please describe the procedure and/or tests that need to be performed to verify these changes. Also, include test configuration details if relevant.
```

Author: Fixed

- [ ] Your changes have to be tested on hardware
- [ ] You have added unit tests

## Test A
1) Step 1
2) Step 2

## Test B
1) Step 1
2) Step 2

## Roslint Test

> - [ ] You have performed linter tests

Reviewer: @patapufate I'd suggest we move this to the checklist. A `roslint` test should be a must.

Author: Yes. You're right
